src/mechanics/constraintsPde/Constraints.cpp

```diff
@@ -0,0 +1,124 @@
+
+    const Equations& equations = mEquations[dof];
+    int numEquations = equations.size();
+
+    Eigen::SparseMatrix<double> matrix(numEquations, numIndependentDofs);
+
+    for (int iEquation = 0; iEquation < numEquations; ++iEquation)
+    {
+        const auto& equation = equations[iEquation];
+        for (Term term : equation.GetTerms())
+        {
+            double coefficient = term.GetCoefficient();
+            int globalDofNumber = term.GetConstrainedDofNumber();
+            if (globalDofNumber == -1 /* should be NodeSimple::NOT_SET */)
+                throw Exception(__PRETTY_FUNCTION__,
+                                "There is no dof numbering for a node in equation" + std::to_string(iEquation) + ".");
+
+            if (globalDofNumber >= numIndependentDofs)
+            {
+                // This corresponds to the last block of size numDependentDofs x numDependentDofs that should be an
+                // identity matrix
+                int transformedDof = globalDofNumber - numIndependentDofs;
+                if (transformedDof != iEquation)
+                    throw Exception(__PRETTY_FUNCTION__, "The numbering of the dependent dofs "
+                                                         "is not in accordance with the equation numbering.");
+                continue; // Do not put the value into the matrix.
+            }
+
+            if (std::abs(coefficient) > 1.e-18)
+                matrix.coeffRef(iEquation, globalDofNumber) = coefficient;
+        }
+    }
+    return matrix;
+}
+
+int Constraints::GetNumEquations(DofType dof) const
+{
+    if (not mEquations.Has(dof))
+        return 0;
+    return mEquations[dof].size();
+}
+
+const Equation& Constraints::GetEquation(DofType dof, int equationNumber) const
+{
+    return mEquations[dof].at(equationNumber);
+}
+
+template <typename T>
+bool Contains(const T& container, NuTo::ConstraintPde::Term t)
+{
+    return container.find(t) != container.end();
+}
+
+bool Constraints::TermChecker::TermCompare::operator()(const Term& lhs, const Term& rhs) const
+{
+    if (&lhs.GetNode() != &rhs.GetNode())
+        return &lhs.GetNode() < &rhs.GetNode();
+    return lhs.GetComponent() < rhs.GetComponent();
+}
+
+void Constraints::TermChecker::CheckEquation(Equation e)
+{
+    // The new dependent term must not constrain the same dof as any existing terms
+    Term newDependentTerm = e.GetTerms()[0];
+
+    if (Contains(mDependentTerms, newDependentTerm))
+        throw Exception(__PRETTY_FUNCTION__, "The dependent dof of the new equation "
+                                             "is already constrained as a dependent dof in another equation.");
+
+    if (Contains(mOtherTerms, newDependentTerm))
+        throw Exception(__PRETTY_FUNCTION__, "The dependent dof of the new equation "
+                                             "is already constrained in another equation.");
+
```

Comment:
Proposition accepted. And the check for the factor to be one is not needed since there is no way to construct an equation with a dependent term factor != 1.